src/resource/resource.ts

```diff
@@ -338,6 +339,27 @@ export function isPodOwner(
   return podOwner === webId;
 }
 
+/**
+ * Extends the regular JavaScript error object with access to the status code and status message.
+ */
+export class FetchError extends Error {
+  public readonly statusCode: number;
+  public readonly statusText: string;
```

Comment:
I would have thought having the entire response object available here might be useful too (e.g. what if the server response contained a detailed explanation of the error (perhaps in RDF) in the body of the response?).

Reply:
Yeah it's listed in the ticket as something to consider, but I preferred to leave it out for now because I'd like to avoid people manually parsing server responses, which will then likely only work with the server they happen to be testing against. If there's something standardised in the responses, we should expose it in a way that will work for every Solid server. (And e.g. the Solid spec should specify that a detail explanation of the error should be included in RDF using some vocabulary.)

Reply:
Hmmm... Ok. Personally I would still have err'ed on the side of providing the full response object. I still consider this entire library fairly low-level, so I'd favor flexibility and access over protecting developers from foot-guns (as that's what higher-level abstractions try to do). But I'm Ok to leave this for now and let's see what happens...

Reply:
Yeah, definitely something we can reconsider when we have higher-level stuff that people use, instead of solid-client directly.